meta-mender-commercial/recipes-mender/mender-gateway/mender-gateway.inc

```diff
@@ -0,0 +1,49 @@
+LICENSE = "CLOSED"
+LICENSE_FLAGS = "commercial"
+
+inherit systemd
+
+SUB_FOLDER_arm = "arm"
+SUB_FOLDER_aarch64 = "aarch64"
+SUB_FOLDER_x86-64 = "x86_64"
+
+COMPATIBLE_HOSTS = "arm|aarch64|x86_64"
+
+FILES_${PN} = " \
+    ${bindir}/mender-gateway \
+"
+
+FILES_${PN}_append_mender-systemd = " \
+    ${systemd_system_unitdir}/mender-gateway.service \
+"
+
+S = "${WORKDIR}/mender-gateway-${PV}"
+
+do_version_check() {
+    if ! ${@'true' if d.getVar('MENDER_DEVMODE') else 'false'}; then
+        if ! strings ${WORKDIR}/${SUB_FOLDER}/mender-gateway | fgrep -q "${PN} ${PV}"; then
+            bbfatal "String '${PN} ${PV}' not found in binary. Is it the correct version? Check with --version. Possible candidates: $(strings ${WORKDIR}/${SUB_FOLDER}/mender-binary-delta | grep '${PN} [a-f0-9]')"
```

Comment:
Copy-pasta error: `mender-binary-delta`

Reply:
Which means we could probably add this to a `bbclass` and parametrize it, right?

Reply:
Also, I was about to suggest to use `nm` to check for the symbol, instead of extracting all the strings from the binary.

Which made me realize that the symbol we're trying to set for the version:
`github.com/mendersoftware/mender-gateway/config.Version` does not exist in the binary...

I will add a PR to fix this up :) 

But the suggestion still persides

Reply:
> Which made me realize that the symbol we're trying to set for the version:
github.com/mendersoftware/mender-gateway/config.Version does not exist in the binary...
>
> I will add a PR to fix this up :)

Alf is taking care of this, see acceptance criteria at https://tracker.mender.io/browse/MEN-5435

Reply:
> Copy-pasta error: mender-binary-delta

Thanks

> Which means we could probably add this to a bbclass and parametrize it, right?

Maybe. But it is a bit different in `mender-monitor` than in `mender-binary-delta`. I think I'll leave it for the future...

> Also, I was about to suggest to use nm to check for the symbol, instead of extracting all the strings from the binary

As we don't really have it in the binary as per today I cannot experiment. I think I'll leave it as it is that is already proven working for `mender-binary-delta`.